common/src/main/java/io/druid/data/SearchableVersionedDataFinder.java

```diff
@@ -42,7 +42,7 @@
    *
    * @return A DataDescriptor which matches pattern, is a child of descriptorBase, and is of the most recent "version" at some point during the method execution.
    */
-  DataDescriptor getLatestVersion(DataDescriptor descriptorBase, @Nullable final Pattern pattern);
+  DataDescriptor getLatestVersion(DataDescriptor descriptorBase, @Nullable Pattern pattern);
```

Comment:
not sure why final is removed here?

Reply:
From https://github.com/checkstyle/checkstyle/issues/45,

> Interface and abstract methods with final parameters have no meaning since they have no implementation and thus the non-final parameters cannot be modified.

Well, I'm not sure about this because I think it is still meaningful to remind developers that variables are final, but this is the way checkstyle does.

Actually, the ```final``` parameters in interface methods has no meaning except for documentation because it has no effect on the generated byte code (https://stackoverflow.com/questions/5380177/final-arguments-in-interface-methods-whats-the-point).